operator/watchers/cilium_endpoint.go

```diff
@@ -147,11 +148,11 @@ func convertToCiliumEndpoint(obj interface{}) interface{} {
 			},
 		}
 		*concreteObj = cilium_api_v2.CiliumEndpoint{}
-		return p
+		return p, nil
 	case cache.DeletedFinalStateUnknown:
 		ciliumEndpoint, ok := concreteObj.Obj.(*cilium_api_v2.CiliumEndpoint)
 		if !ok {
-			return obj
+			return nil, fmt.Errorf("unknown object type %v", concreteObj.Obj)
```

Comment:
```suggestion
			return nil, fmt.Errorf("unknown object type %T", concreteObj.Obj)
```

Please apply this to all other instances